### PR TITLE
[Merged by Bors] - chore: remove tactics

### PR DESCRIPTION
**Pull request by adomani**
More tactics that are not used, found using the linter at #11308.

The PR consists of tactic removals, whitespace changes and replacing a porting note by an explanation.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fin/SuccPred.lean [hunk 1]
```diff
@@ -21,7 +21,7 @@ to a specific `Fin` instance.
 namespace Fin
 
 instance : ∀ {n : ℕ}, SuccOrder (Fin n)
-  | 0 => by constructor <;> first | assumption | intro a; exact elim0 a
+  | 0 => by constructor <;> first | intro a; exact elim0 a
```
Comment (YaelDillies) [inline Mathlib/Data/Fin/SuccPred.lean:24]: Maybe you also need a "single tactic `first`" linter
```suggestion
  | 0 => by constructor <;> exact elim0
```

File: Mathlib/Data/Finset/Lattice.lean [hunk 2]
```diff
@@ -1383,7 +1383,7 @@ theorem max_of_nonempty {s : Finset α} (h : s.Nonempty) : ∃ a : α, s.max = a
 theorem max_eq_bot {s : Finset α} : s.max = ⊥ ↔ s = ∅ :=
   ⟨fun h ↦ s.eq_empty_or_nonempty.elim id fun H ↦ by
       obtain ⟨a, ha⟩ := max_of_nonempty H
-      rw [h] at ha; cases ha; done, -- Porting note: error without `done`
+      rw [h] at ha; cases ha; , -- Porting note: error without `done`
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Lattice.lean:1386]: As before, I'm not sure you should be touching this if you're not solving the porting note

Comment (adomani) [reply Mathlib/Data/Finset/Lattice.lean:1386]: I think that the porting note is simply pointing out that a `,` does not stop the `cases` pattern.  This is the [Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/error.20puzzle/near/426043583).  So, I am going to remove the porting note!

**Comment by adomani** [comment]
!bench

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**